meorg_client/client.py

```diff
@@ -288,40 +290,27 @@ def upload_files(
         responses = list()
         if n == 1:
             for fp in tqdm(files, total=len(files)):
-                response = self._upload_file(fp, attach_to=attach_to)
-                responses.append(response)
+                response = self._upload_file(fp, id=id)
+                responses += response
         else:
-
-            # Disable the auto attach to avoid race condition
-            responses = self._upload_files_parallel(
-                files, n=n, attach_to=None, progress=progress
+            responses += self._upload_files_parallel(
+                files, n=n, id=id, progress=progress
             )
 
-            if attach_to:
-
-                file_ids = list()
-
-                for response in responses:
-                    file_id = response.get("data").get("files")[0].get("file")
-                    file_ids.append(file_id)
-
-                self.attach_files_to_model_output(id=attach_to, files=file_ids)
-
-        return mu.ensure_list(responses)
+        # return mu.ensure_list(responses)
+        return responses
 
     def _upload_file(
```

Comment:
`_upload_files` was capable of uploading multiple files in the one payload, however, this was causing issues with return types (lists vs single objects) so it was simpler to make it the singular case. This is fine as the packet limit on the upload basically prevents uploading multiple files of any meaningful size anyway.

TL;DR - making this single-file made everything a lot simpler for the other methods which all call this.